docs/docs/dev/debug/index.md

```diff
@@ -0,0 +1,5 @@
+The xemu source tree can be quite intimidating at first due to its qemu roots, but the majority of Xbox-specific code is confined to the `xemu/hw/xbox/` folder where the rest can be disregarded.
+
+The basic principles are to emulate an original Xbox system at the hardware level via qemu intercepting memory-mapped IO (MMIO) intended for devices and re-implementing functionality as close to the original as possible. In the case of the NV2A GPU, these MMIO get translated to equivalent OpenGL instructions.

```

Comment:
The gist is correct but the terminology is a bit loose.

`qemu intercepting` really means CPU emulation (which is provided by the QEMU project with improvements made by xemu project for Xbox emulation) properly routes memory accesses made to emulated hardware.

MMIO here means device access by CPU client via memory address space, which does happen for hardware configuration. However pushbuffer commands are actually consumed and executed by (emulated) hardware asynchronously via DMA. Then when processing those commands we emulate the behavior of the 2D/3D graphics hardware using OpenGL API on your host machine. There are parallels between hardware commands and OpenGL API, but it is not just a simple 1:1 map, which was my interpretation of the wording.

Let's re-word it for clarity/simplify, or we can just drop this paragraph for now and add details later.